Update to I2C docs: Clarify address expectation and return values (#8413)
qmk/qmk_firmware: docs/i2c_driver.md

The I2C Master drivers used in QMK have a set of common functions to allow portability between MCUs.

## An important note on I2C Addresses

All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`.

You can either do this on each call to the functions below, or once in your definition of the address.  For example if your device has an address of `0x18`:

> `#define MY_I2C_ADDRESS 0x18 << 1`

Reviewer: ```suggestion
`#define MY_I2C_ADDRESS (0x18 << 1)`
```

The I2C Master drivers used in QMK have a set of common functions to allow portability between MCUs.

## An important note on I2C Addresses

All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`

> 

Reviewer 2: ```suggestion
All of the addresses expected by this driver should be pushed to the upper 7 bits of the address byte.  Setting
the lower bit (indicating read/write) will be done by the respective functions.  Almost all I2C addresses listed 
on datasheets and the internet will be represented as 7 bits occupying the lower 7 bits and will need to be
shifted to the left (more significant) by one bit.  This is easy to do via the bitwise shift operator `<< 1`.
```

You can either do this on each call to the functions below, or once in your definition of the address.  For example if your device has an address of `0x18`:

`#define MY_I2C_ADDRESS 0x18 << 1`

See: 

https://www.robot-electronics.co.uk/i2c-tutorial 

> For more information about I2C addressing and other technial details.

Reviewer 2: ```suggestion
See https://www.robot-electronics.co.uk/i2c-tutorial for more information about I2C addressing and other technical details.
```

|`i2c_status_t i2c_readReg(uint8_t devaddr, uint8_t regaddr, uint8_t* data, uint16_t length, uint16_t timeout);`        |Same as the `i2c_receive` function but `regaddr` sets from where in the slave the data will be read.                                                                         |
|`i2c_status_t i2c_stop(void);`                                                                                         |Ends an I2C transaction.                                                                                                                                                     |

### Function Return

All the above functions, except `void i2c_init(void);` return the following truth table:

|Return Constant     |Value |Description                                 |
|--------------------|---------------------------------------------------|
|I2C_STATUS_SUCCESS  |  0   |Operation executed successfully.            |
|I2C_STATUS_ERROR    | -1   |Operation failed.                           |

> |I2C_STATUS_TIMEOUT  | -2   |Operation timed out.                        |

Reviewer: ```suggestion
|Return Constant     |Value|Description                     |
|--------------------|-----|--------------------------------|
|`I2C_STATUS_SUCCESS`|0    |Operation executed successfully.|
|`I2C_STATUS_ERROR`  |-1   |Operation failed.               |
|`I2C_STATUS_TIMEOUT`|-2   |Operation timed out.            |
```

These are defined in `i2c_master.h` and it is suggestd you use the definitions for future compatibility if
they change.

> 

Reviewer: ```suggestion
```

|-1             |Operation failed.                                  |
|-2             |Operation timed out.                               |

These are defined in `i2c_master.h` and it is suggested you use this definitions for future compatibility if
they change.

```
#define I2C_STATUS_SUCCESS (0)
#define I2C_STATUS_ERROR (-1)

> #define I2C_STATUS_TIMEOUT (-2)

Reviewer: IMO these names should be part of the above table.
